Fish forces REVIEW NEEDED

reinthal opened the pull request:
Add fish forces to velocity update

Commit:
added .idea/ to .gitignore

Commit:
Merge branch 'master' of https://github.com/fredrikm/SoCSProject16 into fish-forces

Commit:
Add fish forces

Commit:
Merge remote-tracking branch 'origin/fish-forces' into fish-forces

# Conflicts:
#	.gitignore

Commit:
add coloumb fource

Commit:
added force to velocity calculation

Commit:
added force to velocity equation

rasake commented on code/agents.py:
```diff
@@ -64,13 +65,30 @@ def  __init__(self, position, velocity, fish_id, environment, ann, image = None,
+
+        return f
+
+    def get_neighbours(self):
```
Good move to have this in dedicated function, improves readability. But change name to find_neighbours, there are no getters in Python.

rasake commented on code/agents.py:
```diff
@@ -64,13 +65,30 @@ def  __init__(self, position, velocity, fish_id, environment, ann, image = None,
+
+    def get_fish_forces(self):
+        f = np.array([0.0, 0.0])
+        neighbours = self.get_neighbours()
```
Assume neigbours are already found by think(), just get them directly from the attribute self.neighbouring_fish. That way we don't have to do the search twice.

rasake commented on code/agents.py:
```diff
@@ -64,13 +65,30 @@ def  __init__(self, position, velocity, fish_id, environment, ann, image = None,
     """
-    
+
+    def get_fish_forces(self):
```
Don't use get in python method names. Call it something else like "calculate_fish_forces", or make it a property using the @property decorator.

rasake commented on code/main.py:
```diff
@@ -78,6 +78,9 @@ def __init__(self, nbr_fish, nbr_predators, boundaries, ann_weights, graphics_on
             self.fish_lst.append(fish)
 
+        # Initialize physical constants of  Coloumb force equation
+        self.k = 1
```
Did you intend to put this attributes in each fish instance? Then they must be indented as part of the for loop and it should be fish.k = 1, fish.power=6, in that case I believe.

reinthal replied:
No. It's part of the environment. Makes more sense that physics is part of the environment right?

rasake replied:
I see.

Commit:
Merge remote-tracking branch 'origin/fish-forces' into fish-forces

# Conflicts:
#	code/agents.py

Commit:
Fix java-like code

rasake commented on code/agents.py:
```diff
@@ -64,13 +65,32 @@ def  __init__(self, position, velocity, fish_id, environment, ann, image = None,
+
+            k = self.environment.k
+            power = self.environment.power
+            f_i = k / np.abs(fish.position-self.position)**power
```
f_i should be directed as rel. distanc vector between fish and have magnitude inversely proport. to (norm of rel. distance)^power

Commit:
fix Obvious bug

fix obvious bug and fine tune parameters.

With these parameter settings fish do not want to swim into each other
at all. Might have to lower this later during optimization. power = 2
not 6 and k = 10.

kognitiva commented:
@rasake I fixed the fish forces now! I set the parameters very high to make sure that it was doing its job. I guess we have to run some optimization before we know what we level of force we need to get nice results.

rasake commented:
Funkar kraften nu?

Commit:
Fix merge conflicts. Fine-tune parameters for forces. Fix equation for coloumb force f / m = a
